### PR TITLE
[TVMScript] Represent ramp as index slice

### PR DESCRIPTION
**Pull request by wrongtest**
Hi there, the PR propose to represent `tir.RampNode` in buffer access index just as python slice. I believe this could be a more user friendly way for script users on SIMD-based target, since it just mimic usages in various python numerical libraries like numpy.

For example:
`A[T.ramp(0, 1, 1024)]`  => `A[0:1024]`
`A[T.ramp(i*32, 1, 32)]` => `A[i*32:i*32+32]`
`A[T.ramp(i*4, 4, 32)]` => `A[i*4:i*4+128:4]`

The syntax sugar is only available for constant stride and lanes. Also please correct me on two points:
- TIR lowering actually do not support vectorized index in non-last axes, but it is not checked by current change.
- `Slice` with non-trivial step is only allowed in buffer indices, we may want to ensure all other usages should check step==1.

cc @junrushao1994 @Hzfengsy @Lunderberg

### CODE REVIEW

File: tests/python/unittest/test_tvmscript_error_report.py [hunk 1]
```diff
@@ -658,5 +635,42 @@ def test_preflattened_buffer_map_offset_factor():
...
+def strided_buffer_region(A: T.handle):
+    # do not allow stride in buffer region
+    A = T.match_buffer((128, 128), "int32")
+    with T.block():
+        T.reads([])
+        T.writes([A[0:128:2, 0:128:3]])  # error
+        T.evaluate(T.call_extern("strided_compute", dtype=""))
+
+
+def strided_buffer_region(A: T.handle):
```
Comment (Hzfengsy) [inline tests/python/unittest/test_tvmscript_error_report.py:647]: the duplicated test

File: tests/python/unittest/test_tvmscript_error_report.py [hunk 2]
```diff
@@ -658,5 +635,42 @@ def test_preflattened_buffer_map_offset_factor():
...
+
+
+def access_reversed_slice(A: T.handle):
+    # do not allow reversed slice step
+    A = T.match_buffer((128,), "int32")
+    A[0:128:-1] = T.broadcast(1, 128)
+
+
+def access_non_const_slice(A: T.handle):
+    # do not allow reversed slice step
```
Comment (Hzfengsy) [inline tests/python/unittest/test_tvmscript_error_report.py:663]: error comment?

**Review by Hzfengsy — APPROVED** [review]
Overall LGTM

**Comment by junrushao1994** [comment]
CC @yelite would you mind taking a look as well?

**Review by yelite — APPROVED** [review]

**Review by junrushao1994 — APPROVED** [review]

**Pull request merged**